contribute.md

```diff
@@ -3,12 +3,246 @@ layout: page
 title: Contribute
 ---
 
-Each subsystem is separated into individual repositories within the linux-system-roles GitHub project. Just open a new issue against the appropriate subsystem’s issue tracker to report bugs or request enhancements at the subsytem listed in
+Each subsystem is separated into individual repositories within the
+[linux-system-roles](https://github.com/linux-system-roles) GitHub project.
+Just open a new issue against the appropriate subsystem's issue tracker to
+report bugs or request enhancements.  New subsystem requests or feedback can be
+provided to the project's landing page at
+[linux-system-roles.github.io](https://linux-system-roles.github.io) Pull
+requests welcome!
 
-[https://github.com/linux-system-roles](https://github.com/linux-system-roles)
 
-New subsystem requests or feedback can be provided to the project’s landing page at linux-system-roles.github.io Pull requests welcome!
+## Code structure
+The repository is structured as described in [Ansible Roles documentation](https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#role-directory-structure)
 
-The subsystems are licensed under different licences. By submitting a patch for inclusion, you are agreeing to license your changes under the subsystem licence.
+In addition to the standard roles files described in the roles documentation ,there may be some addition directoris like 
 
-For discussions and questions, please use our mailing list (archives) or join our IRC channel #systemroles on freenode.
+- `./examples/` - Contains YAML examples for different configurations.
+
+- `./tests/tests_*.yml` - Contains the complete tests for the role.`./tests/tasks/` 
+  contains task snippets that are used in multiple tests to avoid having the same 
+  code repeated multiple times.  
+  Note that for the `network` role,`./tests/tests_*.yml` are shims to run tests 
+  once for every provider and `./tests/playbooks/` contain the actual tests. 
+
+The rest of files in the root folder mostly serve as configuration files for diferent
+testing tools and bots that help with the manteinance of the project.
+
+
+The code files will always have the imports on the first place, followed by constants
+and in the last place, classes and methods. The style of python coding for this project
```

Comment:
"For Python code, in the source file, the imports will generally come first, followed by constants, classes, and methods."